pilot/pkg/networking/util/util.go

```diff
@@ -503,23 +499,21 @@ func BuildLbEndpointMetadata(networkID network.ID, tlsMode, workloadname, namesp
 	// server does not have sidecar injected, and request fails to reach server and thus metadata exchange does not happen.
 	// Due to performance concern, telemetry metadata is compressed into a semicolon separted string:
 	// workload-name;namespace;canonical-service-name;canonical-service-revision;cluster-id.
-	if features.EndpointTelemetryLabel {
```

Comment:
Remove this feature flag completely?

Reply:
The problem is, previously we have all or nothing telemetry - either its enabled mesh wide, or not. Now, it can be enabled per workload. As a result, we don't know for which endpoints we need this label and for which we do not. We need it if the other side doesn't have MX filters, but we have no way to know that. Some bad options:
* use `tlsMode` label as a proxy for "has sidecar" and expect they will add MX. This won't work since its overloading `tlsMode` which may not always equal "has sidecar"
* Add a new label for "has sidecar" - avoided for complexity
* Check if any Telemetry exists, only add if so - doesn't actually work for multicluster, since we cannot know the full config of all pods in other clusters

So this will result in no change for any cluster using telemetry (API or Envoyfilter), but a moderate EDS perf cost when telemetry is completely disabled. This extra cost is not ideal, but I think better than the alternatives.

We could keep the flag and not set it in `manifests/` automatically I suppose, but it doesn't seem like something we want to keep around long term?

WDYT

Reply:
>  This extra cost is not ideal, but I think better than the alternatives.

Yeah I agree, we probably should just remove this flag. I think most large deployment should have telemetry enabled, and swallowed the performance impact. Always adding these metadata makes sense to me.